### PR TITLE
Add endpoint GET /api/companies/ for the ListCompanies feature.

### PR DESCRIPTION
**Pull request by laurandarius**

### CODE REVIEW

**Comment by laurandarius** [comment]
[Add endpoint GET /api/companies/ for the ListCompanies feature.](https://trello.com/c/NMp7R7Zt/19-add-endpoint-get-api-companies-for-the-listcompanies-feature)

File: app/repositories/company.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
+const db = require('../models');
 const Repository = require('./crudRepository');
 
-class CompanyRepository extends Repository {};
+class CompanyRepository extends Repository {
+  async getAll() {
+    return db.companies.findAll({ raw: true});
```
Comment (lgnstn) [inline app/repositories/company.js]: Missing space before `}`.

File: app/services/company.js [hunk 2]
```diff
@@ -72,6 +72,12 @@ class CompanyService extends CrudService {
 
     return deleted;
   }
+
+  async getAll() {
+    const found = await companyRepository.getAll();
```
Comment (lgnstn) [inline app/services/company.js]: You can directly return this.

**Review by lgnstn — CHANGES_REQUESTED** [review]

File: app/repositories/company.js (hunk 1 above)
Comment (lgnstn) [reply app/repositories/company.js]: I think it is best to add this method here https://github.com/EdiOancea/ferret-api/blob/master/app/services/crudService.js

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by laurandarius** [comment]
@lgnstn updated.

File: app/routes/company.js [hunk 3]
```diff
@@ -1,3 +1,17 @@
-const crudRouter = require('./crudRouter');
+const express = require('express');
```
Comment (lgnstn) [inline app/routes/company.js]: Why?

File: tmp/.gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
```
Comment (lgnstn) [inline tmp/.gitignore]: Can't you use the global git ignore file?

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by laurandarius** [comment]
@lgnstn updated.

File: app/controllers/company.js [hunk 5]
```diff
@@ -1,3 +1,10 @@
 const CrudController = require('./crudController');
+const companyService = require('../services/company');
 
-module.exports = new CrudController('company');
+class CompanyController extends CrudController {
+  async getAll(req, res, next) {
```
Comment (lgnstn) [inline app/controllers/company.js]: Please add the `getAll` method in `CrudController`.

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by laurandarius** [comment]
@lgnstn updated again.

**Review by lgnstn — APPROVED** [review]

**Pull request merged**